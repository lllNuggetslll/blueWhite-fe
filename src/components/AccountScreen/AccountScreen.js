import React from "react";
import { object } from "prop-types";
import { get } from "lodash";
import styled from "styled-components";
import Avatar from "@material-ui/core/Avatar";

const Container = styled.div``;

const AvatarContainer = styled.div`
  display: flex;
  justify-content: center;
`;

const BalanceContainer = styled.div`
  display: flex;
  align-items: center;
`;

const Balance = styled.div`
  background-color: lightgrey;
  width: 300px;
  margin-left: 20px;
  border-radius: 3px;
`;

const AccountScreen = ({ userInfo }) => {
  const balance = get(userInfo, "balance") || 0;

  const initials = () => {
    const { first, last } = get(userInfo, "name", { first: " ", last: " " });

    return first[0] + last[0] || "";
  };

  return (
    <Container>
      <h1>Your Account</h1>
      <AvatarContainer>
        <Avatar>{initials()}</Avatar>
      </AvatarContainer>
      <BalanceContainer>
        <h3>Balance:</h3>
        <Balance>${balance}</Balance>
      </BalanceContainer>
    </Container>
  );
};

AccountScreen.propTypes = {
  userInfo: object.isRequired
};

export default AccountScreen;
