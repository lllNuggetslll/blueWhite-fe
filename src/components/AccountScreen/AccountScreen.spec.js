import React from "react";
import AccountScreen from "./AccountScreen";
import { shallow } from "enzyme";

describe("Account Screen", () => {
  const wrapper = shallow(<AccountScreen userInfo={{}} />);

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
