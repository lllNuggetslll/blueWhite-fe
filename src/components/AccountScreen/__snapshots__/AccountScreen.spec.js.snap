// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Screen should render 1`] = `
<styled.div>
  <h1>
    Your Account
  </h1>
  <styled.div>
    <WithStyles(ForwardRef(Avatar))>
        
    </WithStyles(ForwardRef(Avatar))>
  </styled.div>
  <styled.div>
    <h3>
      Balance:
    </h3>
    <styled.div>
      $
      0
    </styled.div>
  </styled.div>
</styled.div>
`;
